plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-variations-controller.php

```diff
@@ -485,6 +485,23 @@ protected function set_variation_image( $variation, $image ) {
 		return $variation;
 	}
 
+	/**
+	 * Get object.
+	 *
+	 * @since  3.0.0
+	 * @param  int $id Object ID.
+	 * @return WC_Data
+	 */
+	protected function get_object( $id ) {
+		/**
+		 * Filter the product object that is fetched in the context of the REST API.
+		 *
+		 * @param WC_Product $product The product.
+		 * @since 9.0.0
+		 */
+		return apply_filters( 'woocommerce_rest_get_product', wc_get_product( $id ) );
```

Comment:
> so instead of sending the template id through the meta data, maybe we can just add it to the root? But that is not a must by any means.

I think this is a big change, as it's already consumed through meta_data in existing code in the front-end, and it is currently stored in postmeta. I would opt to not do this, at least right now.

Reply:
> I think this is a big change, as it's already consumed through meta_data in existing code in the front-end, and it is currently stored in postmeta. I would opt to not do this, at least right now.

Makes sense